src/plots/ternary/ternary.js

```diff
@@ -156,6 +156,9 @@ proto.updateLayers = function(ternaryLayout) {
             } else if(d === 'grids') {
                 grids.forEach(function(d) {
                     layers[d] = s.append('g').classed('grid ' + d, true);
+
+                    var fictID = (d === 'bgrid') ? 'x' : 'y';
+                    layers[d].append('g').classed(fictID, true);
```

Comment:
why do we need this now?

Reply:
Yeah, that was a bit of a hack, we could probably find a way to clean it up for derived axes like this. When I got rid of `overgrid` and put everything in `gridlayer`, it meant nesting inside `gridlayer` and [selecting](https://github.com/plotly/plotly.js/pull/2227/files/2c9d90fd26952b91290540c35ef284f9aa6d90ed..3625be4e516376ff4ce65f0c0223d1d9b99c2d24#diff-f6d5ed25a5d6f57c70063e1e1bbc407dR2199) that nesting later.

Reply:
Ok thanks for the info. I'll need to hack in something similar for polar. Maybe I'll try to clean it up :ok_hand: